point_cloud_client/src/lib.rs

```diff
@@ -28,40 +28,43 @@ impl PointCloudClient {
             .iter()
             .map(|location| data_provider_factory.generate_data_provider(location))
             .collect::<Result<Vec<Box<dyn DataProvider>>>>()?;
+        let mut aabb: Option<Aabb3<f64>> = None;
+        let unite = |bbox: &Aabb3<f64>, to: &mut Option<Aabb3<f64>>| {
+            if to.is_none() {
+                *to = Some(*bbox);
+            } else {
+                *to = to.map(|b| b.union(bbox));
+            }
+        };
         let point_clouds = if data_providers[0].meta_proto()?.has_octree() {
             PointClouds::Octrees(
                 data_providers
                     .into_iter()
-                    .map(|provider| Octree::from_data_provider(provider))
+                    .map(|provider| {
+                        Octree::from_data_provider(provider).map(|octree| {
+                            unite(octree.bounding_box(), &mut aabb);
+                            octree
+                        })
+                    })
                     .collect::<Result<Vec<Octree>>>()?,
             )
         } else {
             PointClouds::S2Cells(
                 data_providers
                     .into_iter()
-                    .map(|provider| S2Cells::from_data_provider(provider))
+                    .map(|provider| {
+                        S2Cells::from_data_provider(provider).map(|s2_cells| {
+                            unite(s2_cells.bounding_box(), &mut aabb);
+                            s2_cells
+                        })
+                    })
                     .collect::<Result<Vec<S2Cells>>>()?,
             )
         };
-        let mut aabb = Aabb3::zero();
-        match &point_clouds {
-            PointClouds::Octrees(octrees) => {
-                if let Some((first_octree, remaining_octrees)) = octrees.split_first() {
-                    aabb = *first_octree.bounding_box();
-                    for octree in remaining_octrees {
-                        aabb = aabb.union(octree.bounding_box());
-                    }
-                }
-            }
-            PointClouds::S2Cells(_s2_cells) => {
-                // TODO(feuerste): Add bounding box to s2 meta.
-                unimplemented!();
-            }
-        }
 
         Ok(PointCloudClient {
             point_clouds,
-            aabb,
+            aabb: aabb.unwrap_or_else(Aabb3::zero),
```

Comment:
in ECEF is the null bounding box in the center of the earth, could this bring some errors in the Point cloud client if it is defined like this? should be better this remain undefined?

Reply:
Previously it was the same for ECEF octrees. Actually this case should never happen now anyway, since I check in the beginning, whether `locations.is_empty()`.